samples/bluetooth/mesh/src/main.c

```diff
@@ -150,6 +291,75 @@ static const struct bt_mesh_prov prov = {
 	.reset = prov_reset,
 };
 
+/** Send an OnOff Set message from the Generic OnOff Client to all nodes. */
+static int gen_onoff_send(bool val)
+{
+	struct bt_mesh_msg_ctx ctx = {
+		.app_idx = models[3].keys[0], /* Use the bound key */
+		.addr = BT_MESH_ADDR_ALL_NODES,
+		.send_ttl = BT_MESH_TTL_DEFAULT,
+	};
+	static uint8_t tid;
+
+	if (ctx.app_idx == BT_MESH_KEY_UNUSED) {
+		printk("The Generic OnOff Client must be bound to a key before "
+		       "sending.\n");
+		return -ENOENT;
+	}
+
+	BT_MESH_MODEL_BUF_DEFINE(buf, OP_ONOFF_SET_UNACK, 2);
+	bt_mesh_model_msg_init(&buf, OP_ONOFF_SET_UNACK);
+	net_buf_simple_add_u8(&buf, val);
+	net_buf_simple_add_u8(&buf, tid++);
+
+	printk("Sending OnOff Set: %s\n", onoff_str[val]);
+
+	return bt_mesh_model_send(&models[3], &ctx, &buf, NULL, NULL);
+}
+
+static void button_pressed(struct k_work *work)
+{
+	if (bt_mesh_is_provisioned()) {
+		(void)gen_onoff_send(!onoff.val);
+		return;
+	}
+
+	/* Self-provision with an arbitrary address.
+	 *
+	 * NOTE: This should never be done in a production environment.
+	 *       Addresses should be assigned by a provisioner, and keys should
+	 *       be generated from true random numbers. It is done in this
+	 *       sample to allow testing without a provisioner.
+	 */
+	uint16_t addr = k_uptime_get_32() & BIT_MASK(15);
+	static uint8_t net_key[16];
+	static uint8_t dev_key[16];
+	static uint8_t app_key[16];
+	int err;
+
+	printk("Self-provisioning with address 0x%04x\n", addr);
```

Comment:
Why not use the low two bytes of the device id, so it's consistent with what a provisioner would use?

Reply:
Good point, I'll change this